src/rocker/extensions.py

```diff
@@ -273,3 +273,26 @@ def register_arguments(parser, defaults={}):
     def check_args_for_activation(cls, cli_args):
         """ Returns true if the arguments indicate that this extension should be activated otherwise false."""
         return True if cli_args.get('env') or cli_args.get('env_file') else False
+
+
+class Privileged(RockerExtension):
+    """Add the --privileged option to docker arguments."""
+    @staticmethod
+    def get_name():
+        return 'privileged'
+
+    def __init__(self):
+        self.name = Privileged.get_name()
+
+    def get_snippet(self, cli_args):
+        return ''
+
+    def get_docker_args(self, cli_args):
+        return '--privileged'
```

Comment:
This needs to be prepended with whitespace to make sure not to run into a preceding argument if not first. 
```suggestion
        return ' --privileged'
```

They are currently string appended https://github.com/osrf/rocker/blob/4f9f6d7595ccc23a2ae9b600a2262deefb676d89/src/rocker/core.py#L257 This could be made more robust with a ' '.join(...) it looks like other places have this potential problem depending on ordering.

Reply:
Wouldn't it be better to actually let extensions return a list of strings and let core.py manage formatting the command, if needed?

Reply:
I think a single string at least with core.py inserting the space. "If needed" is not clearly resolved if I return multiple.

Reply:
There might be a few words missing in these last sentences and I don't get their sense. Anyway, with "if needed", I meant that `subprocess.run` and partially `pexcpect.spawn` accept a list of strings as argument to launch the command.